### PR TITLE
ENH: make yi_vl support streaming output

### PR DESCRIPTION
**Pull request by Minamiyama**
https://github.com/xorbitsai/inference/assets/918617/d8e5d541-a2b5-4b5f-8858-a06448298c8e

### CODE REVIEW

**Comment by qinxuye** [comment]
Could you rebase the main branch which can pass the CI?

**Comment by Minamiyama** [comment]
> Could you rebase the main branch which can pass the CI?

merged😄

File: xinference/model/llm/pytorch/yi_vl.py [hunk 1]
```diff
@@ -229,25 +199,72 @@ def chat(
...
+            c = self._generate(streamer, stop_str)
+            return self._to_chat_completion(c)
+
+    def _generate(self, streamer, stop_str) -> Completion:
         generated_text = ""
         for new_text in streamer:
             generated_text += new_text
             if generated_text.endswith(stop_str):
                 generated_text = generated_text[: -len(stop_str)]
-        r = self._parse_text(generated_text)
```
Comment (qinxuye) [inline xinference/model/llm/pytorch/yi_vl.py:237]: Why do you remove the function?

Comment (Minamiyama) [reply xinference/model/llm/pytorch/yi_vl.py:237]: yes，I found it is mainly used for gradio ui, but not for a general  purpose, I referenced from official code like: 

https://github.com/01-ai/Yi/blob/main/VL/openai_api.py

the parse_text is defined in web_demo:

https://github.com/01-ai/Yi/blob/main/VL/web_demo.py

Comment (qinxuye) [reply xinference/model/llm/pytorch/yi_vl.py:237]: Will it effect the result from gradio?

Comment (Minamiyama) [reply xinference/model/llm/pytorch/yi_vl.py:237]: and the similar codes is also appeared in web_demo.py of qwen_vl, but not in xinference qwen_vl.py:

image


so I thought it's not neccessary for low-level code

Comment (qinxuye) [reply xinference/model/llm/pytorch/yi_vl.py:237]: OK, maybe we should move the code into chat_interface anyway.

Comment (Minamiyama) [reply xinference/model/llm/pytorch/yi_vl.py:237]: > Will it effect the result from gradio?

I tried times, no effect to normal usage, but not sure for special usage, and also asked ChatGPT for analysing the purpose of _parse_for, it seems to process the code reference like markdown format output

File: xinference/model/llm/pytorch/yi_vl.py [hunk 2]
```diff
@@ -229,25 +199,72 @@ def chat(
         t = Thread(target=self._model.generate, kwargs=generate_kwargs)
         t.start()
 
+        if stream:
+            it = self._generate_stream(streamer, stop_str)
+            return self._to_chat_completion_chunks(it)
```
[the remark this answers is not in the document]
Comment (Minamiyama) [reply xinference/model/llm/pytorch/yi_vl.py:204]: 😄just keep same style as it in core.py⬇️（Code freakishness or OCD🤣）

image

[the remark this answers is not in the document]
Comment (qinxuye) [reply xinference/model/llm/pytorch/yi_vl.py:204]: OK to me.

**Review by qinxuye — APPROVED** [review]
LGTM

**Pull request merged**